packages/mobx-state-tree/src/core/node/scalar-node.ts

```diff
@@ -74,36 +63,54 @@ export class ScalarNode implements INode {
             const newPath = subpath === null ? "" : subpath
             if (this.subpath !== newPath) {
                 this.subpath = newPath
+                this.escapedSubpath = escapeJsonPath(this.subpath)
+                this.subpathAtom.reportChanged()
```

Comment:
Good catch! I wonder how did it work prior this? ) I guess reactions for splicing arrays of scalar nodes didn't fire? 
Should we add test for it?

Reply:
how to test this, making sure the scalar node ref didn't change after an applySnapshot?

Reply:
@k-g-a but getPath does not work over strings for example, I might be missing something...

Reply:
could you write the unit test for this? 🙏